fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StringLikeLiteral.java

```diff
@@ -46,11 +46,12 @@ public double getDouble() {
      * get double value
      */
     public static double getDouble(String str) {
+        byte[] bytes = str.getBytes();
         long v = 0;
         int pos = 0;
-        int len = Math.min(str.length(), 7);
+        int len = Math.min(bytes.length, 7);
```

Comment:
any similar problem for varchar like literal?

Reply:
@xzj7019 StringLikeLiteral.java is base class for all character type